modules/copy-template/pkg/templates/template-provider.go

```diff
@@ -70,6 +74,16 @@ func (t *TemplateCreator) CopyTemplate() (*v1.Template, error) {
 
 	log.Logger().Debug("Original template metadata", zap.Any("ObjectMeta", template.ObjectMeta))
 
+	if isCommonTemplate(template) {
+		vm, vmIndex, err := zutils.DecodeVM(template)
```

Comment:
Decoding the template into `VirtualMachine` struct can fail, because the object may not be a valid VM.
Template can contain parameters in any field. These parameters are strings, but they can be in a place, where the decoder expects a different type.

Instead, you can use `Unstructured` to manipulate labels and annotations.